### DISCUSSION TITLE
Clarity core's treeview component - When can this be released ?

### DISCUSSION THREAD

**Discussion by gayathrirajendran** [vmware/clarity#6415] [General] [answered]
Can you let know when a release version of cds core with accessible tree view component (in beta) be made available ? Planning for a migration from angular to core and so interested in the actual release. 
https://github.com/vmware/clarity/issues/6332
@Jinnie

**Comment by mathisscott** [comment, marked as answer]
Tree view has been available in Core since v5.5.0. The public API has been stable for several months now. The only delay to promoting it out of beta is adding documentation to the website. [You can find development documentation here to start using the component](https://clarity.design/storybook/core/?path=/story/components-tree-view--page).

**Reply by gayathrirajendran** [reply]
Expand and collapse of tree nodes isn't working...In the stories the checkbox variant of tree view alone has the expand and collapse of it working. Is this an item still under development ?
Doesnt work: https://clarity.design/storybook/core/?path=/story/components-tree-view--page
Works: https://clarity.design/storybook/core/?path=/story/stories-tree-view--interactive
@mathisscott

**Reply by mathisscott** [reply]
@gayathrirajendran 

The expand/collapse doesn't happen automatically. The Core components follow reactive principles and are largely stateless.

What this means is that the component will communicate back to the app that a user has performed some action that wants the tree node to expand or collapse. Your app can then follow through on that by setting the tree node's `expanded` property to `true` or `false`. 

Here is an example in code: https://github.com/vmware/clarity/blob/263c7ce328b4cbcb34e745f522830360e8dd0a3f/packages/core/src/tree-view/tree.stories.ts#L268

If you are coming from Angular, `expandedChange` would be like an output. If you are using the `@cds/angular` wrappers, that is exactly what it is.
